### PR TITLE
docs: improved documentation for examples

### PR DESCRIPTION
**Pull request by raresgaia123**
added output snippets and explanations
this will ease the understanding on how each example works

## Description

Please provide a meaningful description of what this change will do, or is for. Bonus points for including links to
related issues, other PRs, or technical references.

Note that by _not_ including a description, you are asking reviewers to do extra work to understand the context of this
change, which may lead to your PR taking much longer to review, or result in it not being reviewed at all.

## Type of Change

- [ ] Bug Fix
- [ ] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [x] Documentation
- [ ] Other (please describe)

## Checklist

- [ ] I have read the [contributing guidelines](/CONTRIBUTING.md)
- [ ] Existing issues have been referenced (where applicable)
- [ ] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [ ] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [ ] All new and existing tests pass

### CODE REVIEW

File: examples/all_gather/README.md [hunk 1]
```diff
@@ -35,3 +35,48 @@ python m8d.py --backend nccl --worldinfo 2,2
...
+| 1           | <span style="color: green">tensor([4., 4., 4.], device='cuda:1')</span>| [<span style="color: red">tensor([1., 5., 5.], device='cuda:2')</span>, <span style="color: green">tensor([4., 4., 4.], device='cuda:2')</span>, <span style="color: blue">tensor([2., 2., 5.], device='cuda:2')</span>] |
+| 2           | <span style="color: blue">tensor([2., 2., 5.], device='cuda:2')</span> | [<span style="color: red">tensor([1., 5., 5.], device='cuda:2')</span>, <span style="color: green">tensor([4., 4., 4.], device='cuda:2')</span>, <span style="color: blue">tensor([2., 2., 5.], device='cuda:2')</span>] |
+
+The same pattern applies to world2.
+
+## Failure case
+
+If something goes wrong in one worker, only the world where the worker belongs will be affected, the other worlds will continue their workload,
+therefore, the way Mutiworld is implemented, it's to prevent errors from spreading accross multiple worlds.
```
Comment (myungjin) [inline examples/all_gather/README.md:81]: Mutiworld -> Multiworld

File: examples/all_reduce/README.md [hunk 2]
```diff
@@ -35,3 +35,48 @@ python m8d.py --backend nccl --worldinfo 2,2
...
+| 1           | <span style="color: green">tensor([7.], device='cuda:1')</span>| tensor([<span style="color: cyan">21.</span>], device='cuda:1') (<span style="color: red">9</span> + <span style="color: green">7</span> + <span style="color: blue">5</span> = <span style="color: cyan">21</span>) |
+| 2           | <span style="color: blue">tensor([5.], device='cuda:2')</span> | tensor([<span style="color: cyan">21.</span>], device='cuda:2') (<span style="color: red">9</span> + <span style="color: green">7</span> + <span style="color: blue">5</span> = <span style="color: cyan">21</span>) |
+
+The same pattern applies to world2.
+
+## Failure case
+
+If something goes wrong in one worker, only the world where the worker belongs will be affected, the other worlds will continue their workload,
+therefore, the way Mutiworld is implemented, it's to prevent errors from spreading accross multiple worlds.
```
Comment (myungjin) [inline examples/all_reduce/README.md:81]: Mutiworld -> Multiworld

Please correct this typo across all readme files.

File: examples/all_gather/README.md [hunk 3]
```diff
@@ -35,3 +35,48 @@ python m8d.py --backend nccl --worldinfo 2,2
...
+| 0           | <span style="color: red">tensor([1., 5., 5.], device='cuda:0')</span>  | [<span style="color: red">tensor([1., 5., 5.], device='cuda:2')</span>, <span style="color: green">tensor([4., 4., 4.], device='cuda:2')</span>, <span style="color: blue">tensor([2., 2., 5.], device='cuda:2')</span>] |
+| 1           | <span style="color: green">tensor([4., 4., 4.], device='cuda:1')</span>| [<span style="color: red">tensor([1., 5., 5.], device='cuda:2')</span>, <span style="color: green">tensor([4., 4., 4.], device='cuda:2')</span>, <span style="color: blue">tensor([2., 2., 5.], device='cuda:2')</span>] |
+| 2           | <span style="color: blue">tensor([2., 2., 5.], device='cuda:2')</span> | [<span style="color: red">tensor([1., 5., 5.], device='cuda:2')</span>, <span style="color: green">tensor([4., 4., 4.], device='cuda:2')</span>, <span style="color: blue">tensor([2., 2., 5.], device='cuda:2')</span>] |
+
+The same pattern applies to world2.
+
+## Failure case
+
+If something goes wrong in one worker, only the world where the worker belongs will be affected, the other worlds will continue their workload,
```
Comment (myungjin) [inline examples/all_gather/README.md:80]:
```suggestion
If something goes wrong in one worker, only the world where the worker belongs will be affected, the other worlds will continue their workload.
```

File: examples/all_gather/README.md (hunk 1 above)
Comment (myungjin) [inline examples/all_gather/README.md:81]:
```suggestion
In other words, Mutiworld prevents errors from spreading accross multiple worlds.
```

Please apply this change across readme files.

**Review by myungjin — COMMENTED** [review]
a few comments left

File: examples/all_gather/README.md [hunk 4]
```diff
@@ -35,3 +35,62 @@ python m8d.py --backend nccl --worldinfo 2,2
...
+
+If something goes wrong in one worker, only the world where the worker belongs will be affected, the other worlds will continue their workload.
+In other words, Mutiworld prevents errors from spreading accross multiple worlds.
+In this case, if, for example rank 1 from world1 fails, ranks from world2 will still continue to perform all_gather operation, together with rank 0 (leader).
+
+The following screenshot demonstrates how errors are handled in multiworld:
+
+<p align="center"><img src="../../docs/imgs/all_gather_error.png" alt="all_gather error handling" width="800" height="300"></p>
+
+Exmplanation:
```
Comment (myungjin) [inline examples/all_gather/README.md:88]: Explanation

File: examples/all_gather/README.md [hunk 5]
```diff
@@ -35,3 +35,62 @@ python m8d.py --backend nccl --worldinfo 2,2
...
+
+<ol>
+    <li>Lead worker (rank 0) is communicating with all the worlds (world 1, world 2)</li>
```
Comment (myungjin) [inline examples/all_gather/README.md:91]: Use markdown syntax for ordered list. Reference: https://www.markdownguide.org/basic-syntax/#ordered-lists.
If html is needed, please add explanation in this thread.

File: examples/all_gather/README.md [hunk 6]
```diff
@@ -35,3 +35,62 @@ python m8d.py --backend nccl --worldinfo 2,2
...
+    <li>Process is killed using keyboard interrupt on rank 2 from world 1</li>
+    <li>The exception is caught by all the workers in the same world (rank 1 in this example)</li>
+    <li>The exception is also caught by the lead worker (rank 0)</li>
+    <li>all_gather operation continues, even one of the worlds failed, lead rank continuing communication with all the remaining worlds (world 2 in this example)</li>
```
Comment (myungjin) [inline examples/all_gather/README.md:95]:
```suggestion
    <li>all_gather operation continues; even after one of the worlds failed, the lead rank continues communication with all the remaining worlds (world 2 in this example)</li>
```

File: examples/all_reduce/README.md [hunk 7]
```diff
@@ -35,3 +35,62 @@ python m8d.py --backend nccl --worldinfo 2,2
...
+
+<p align="center"><img src="../../docs/imgs/all_reduce_error.png" alt="all_reduce error handling" width="800" height="300"></p>
+
+Exmplanation:
+
+<ol>
+    <li>Process is killed using keyboard interrupt on rank 2 from world 2</li>
+    <li>The exception is caught by all the workers in the same world (rank 1 from world 2 in this example)</li>
+    <li>The exception is also caught by the lead worker (rank 0)</li>
+    <li>Lead worker (rank 0) continues communication using all_reduce, even one of the worlds failed (world 2 in this example)</li>
```
Comment (myungjin) [inline examples/all_reduce/README.md:94]:
```suggestion
    <li>The lead worker (rank 0) continues communication using all_reduce, even after one of the worlds failed (world 2 in this example)</li>
```

File: examples/broadcast/README.md [hunk 8]
```diff
@@ -35,3 +35,53 @@ python m8d.py --backend nccl --worldinfo 2,2
...
+
+<p align="center"><img src="../../docs/imgs/broadcast_error.png" alt="broadcast error handling" width="800" height="300"></p>
+
+Exmplanation:
+
+<ol>
+    <li>Process is killed using keyboard interrupt on rank 2 from world 1</li>
+    <li>The exception is caught by all the workers in the same world (rank 1 from world 1 in this example)</li>
+    <li>The exception is also caught by the lead worker (rank 0)</li>
+    <li>Lead worker (rank 0) continues to broadcast tensors, with the remaining worlds (world 2 in this example)</li>
```
Comment (myungjin) [inline examples/broadcast/README.md:85]:
```suggestion
    <li>The lead worker (rank 0) continues to broadcast tensors, with the remaining worlds (world 2 in this example)</li>
```

File: examples/broadcast/README.md [hunk 9]
```diff
@@ -35,3 +35,53 @@ python m8d.py --backend nccl --worldinfo 2,2
...
+    <li>All other workers from all remaining worlds will continue recieve tensors from the lead worker (rank 0)</li>
```
Comment (myungjin) [inline examples/broadcast/README.md:86]:
```suggestion
    <li>All other workers from all remaining worlds will continue to receive tensors from the lead worker (rank 0)</li>
```

File: examples/gather/README.md [hunk 10]
```diff
@@ -35,3 +35,62 @@ python m8d.py --backend nccl --worldinfo 2,2
...
+
+<p align="center"><img src="../../docs/imgs/gather_error.png" alt="gather error handling" width="800" height="300"></p>
+
+Exmplanation:
+
+<ol>
+    <li>Process is killed using keyboard interrupt on rank 2 from world 1</li>
+    <li>The exception is caught by all the workers in the same world (rank 1 from world 1 in this example)</li>
+    <li>The exception is also caught by the lead worker (rank 0)</li>
+    <li>Lead worker (rank 0) continues to gather tensors, from the remaining worlds (world 2 in this example)</li>
```
Comment (myungjin) [inline examples/gather/README.md:94]:
```suggestion
    <li>The lead worker (rank 0) continues to gather tensors, from the remaining worlds (world 2 in this example)</li>
```

File: examples/gather/README.md [hunk 11]
```diff
@@ -35,3 +35,62 @@ python m8d.py --backend nccl --worldinfo 2,2
...
+    <li>All other workers from all remaining worlds will continue send tensors to the lead worker (rank 0)</li>
```
Comment (myungjin) [inline examples/gather/README.md:95]:
```suggestion
    <li>All other workers from all remaining worlds will continue to send tensors to the lead worker (rank 0)</li>
```

File: examples/reduce/README.md [hunk 12]
```diff
@@ -35,3 +35,67 @@ python m8d.py --backend nccl --worldinfo 2,2
...
+
+<p align="center"><img src="../../docs/imgs/reduce_error.png" alt="reduce error handling" width="800" height="300"></p>
+
+Exmplanation:
+
+<ol>
+    <li>Process is killed using keyboard interrupt on rank 2 from world 2</li>
+    <li>The exception is caught by all the workers in the same world (rank 1 from world 2 in this example)</li>
+    <li>The exception is also caught by the lead worker (rank 0)</li>
+    <li>Lead worker (rank 0) continues to be the destination of reduced tensor, from the remaining worlds (world 1 in this example)</li>
```
Comment (myungjin) [inline examples/reduce/README.md:99]:
```suggestion
    <li>The lead worker (rank 0) continues to be the destination of reduced tensor for the remaining worlds (world 1 in this example)</li>
```

File: examples/scatter/README.md [hunk 13]
```diff
@@ -35,3 +35,54 @@ python m8d.py --backend nccl --worldinfo 2,2
...
+
+<p align="center"><img src="../../docs/imgs/scatter_error.png" alt="scatter error handling" width="800" height="300"></p>
+
+Exmplanation:
+
+<ol>
+    <li>Process is killed using keyboard interrupt on rank 1 from world 2</li>
+    <li>The exception is caught by all the workers in the same world (rank 2 from world 2 in this example)</li>
+    <li>The exception is also caught by the lead worker (rank 0)</li>
+    <li>Lead worker (rank 0) continues scatter tensor to the remaining workds (world 1 in this example)</li>
```
Comment (myungjin) [inline examples/scatter/README.md:86]:
```suggestion
    <li>The lead worker (rank 0) continues to scatter tensor to the remaining worlds (world 1 in this example)</li>
```

File: .markdownlint.json [hunk 14]
```diff
@@ -16,7 +16,10 @@
         "allowed_elements": [
             "a",
             "img",
-            "p"
+            "p",
+            "span",
+            "ol",
```
Comment (myungjin) [inline .markdownlint.json:21]: please remove "ol" and "li" if ordered list can be expressed with markdown syntax.

**Review by myungjin — COMMENTED** [review]
Comments are left. Some comments were left only in one readme.md file, but they should be applied in all other readme.md file. So, please take a careful look.

File: examples/all_gather/README.md (hunk 5 above)
Comment (raresgaia123) [reply examples/all_gather/README.md:91]: No reason for using html. I updated using markdown syntax.

**Review by myungjin — APPROVED** [review]

**Pull request merged**